File: backend/ŠišAppApi/Models/ServiceCategory.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ŠišAppApi.Models;

public class ServiceCategory
{
    [Key]
    public int Id { get; set; }
    
    [Required]
    [MaxLength(100)]
    public string Name { get; set; } = string.Empty;
    
    public string? Description { get; set; }
    
    public string? ImageId { get; set; }
    
    [ForeignKey("ImageId")]
    public Image? Image { get; set; }
    
    public int? ParentCategoryId { get; set; }
    
    [ForeignKey("ParentCategoryId")]
    public ServiceCategory? ParentCategory { get; set; }
    
    public int DisplayOrder { get; set; } = 0;
    
    public bool IsActive { get; set; } = true;
    
    [Required]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    
    public DateTime? UpdatedAt { get; set; }
    
    public bool IsDeleted { get; set; } = false;
    
    public DateTime? DeletedAt { get; set; }
    
    // Navigation properties
    public virtual ICollection<ServiceCategory> SubCategories { get; set; } = new List<ServiceCategory>();
    public virtual ICollection<Service> Services { get; set; } = new List<Service>();
} 

File: sisapp_frontend/lib/models/service_category.dart
class ServiceCategory {
  final int id;
  final String name;
  final String? description;
  final String? imageId;
  final int? parentCategoryId;
  final String? parentCategoryName;
  final int displayOrder;
  final bool isActive;

  ServiceCategory({
    required this.id,
    required this.name,
    this.description,
    this.imageId,
    this.parentCategoryId,
    this.parentCategoryName,
    required this.displayOrder,
    this.isActive = true,
  });

  factory ServiceCategory.fromJson(Map<String, dynamic> json) {
    return ServiceCategory(
      id: (json['id'] as num?)?.toInt() ?? 0,
      name: json['name'] ?? '',
      description: json['description'],
      imageId: json['imageId'],
      parentCategoryId: (json['parentCategoryId'] as num?)?.toInt(),
      parentCategoryName: json['parentCategoryName'],
      displayOrder: (json['displayOrder'] as num?)?.toInt() ?? 0,
      isActive: json['isActive'] ?? true,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'imageId': imageId,
      'parentCategoryId': parentCategoryId,
      'parentCategoryName': parentCategoryName,
      'displayOrder': displayOrder,
      'isActive': isActive,
    };
  }
}
